utils/treesitter/src/scanner.c

```diff
@@ -44,6 +45,33 @@ static bool token_allowed_after_binary_operator(char c) {
 
 static bool is_whitespace(char c) { return c == ' ' || c == '\n'; }
 
+static void advance(struct TSLexer* lexer) {
+  lexer->advance(lexer, /* skip= */ false);
+}
+
+static bool eat(struct TSLexer* lexer, const char* string) {
```

Comment:
If this returns `false`, the lexer has advanced an unknown number of tokens , which seems hard to recover from.